onmt/inputters/text_dataset.py

```diff
@@ -163,28 +177,44 @@ def text_fields(**kwargs):
         TextMultiField
     """
 
-    n_feats = kwargs["n_feats"]
+    feats = kwargs["feats"]
     include_lengths = kwargs["include_lengths"]
     base_name = kwargs["base_name"]
     pad = kwargs.get("pad", DefaultTokens.PAD)
     bos = kwargs.get("bos", DefaultTokens.BOS)
     eos = kwargs.get("eos", DefaultTokens.EOS)
     truncate = kwargs.get("truncate", None)
     fields_ = []
-    feat_delim = u"￨" if n_feats > 0 else None
-    for i in range(n_feats + 1):
-        name = base_name + "_feat_" + str(i - 1) if i > 0 else base_name
-        tokenize = partial(
-            _feature_tokenize,
-            layer=i,
-            truncate=truncate,
-            feat_delim=feat_delim)
-        use_len = i == 0 and include_lengths
-        feat = Field(
-            init_token=bos, eos_token=eos,
-            pad_token=pad, tokenize=tokenize,
-            include_lengths=use_len)
-        fields_.append((name, feat))
+
+    feat_delim = None #u"￨" if n_feats > 0 else None
+
+    # Base field
+    tokenize = partial(
+        _feature_tokenize,
+        layer=None,
+        truncate=truncate,
+        feat_delim=feat_delim)
```

Comment:
This part is a bit misleading, `_feature_tokenize` is not actually tokenizing features...
I understand this is mainly for simplicity and modifying less code but maybe add a comment there to explicit what's done.

Reply:
Yes, I did keep that function for simplicity. Currently it only performing truncation of the length. Maybe we could change its name?

Reply:
Sure, something like `_maybe_truncate`?

Reply:
After changing function's name I get the following error whenever an old model is used in either `training_from` or translating. I guess `_feature_tokenize` is stored in the model and therefore we could have backwards compatibility issues...

`AttributeError: Can't get attribute '_feature_tokenize' on <module 'onmt.inputters.text_dataset' from '/OpenNMT-py/onmt/inputters/text_dataset.py'>`

Reply:
There is also a unused import of the function in here:
https://github.com/OpenNMT/OpenNMT-py/blob/baa3d73f645f10e8f3a94b334fc217afdfc4e550/onmt/inputters/inputter.py#L16

Reply:
That's strange. 
What's the full trace before that `AttributeError` ?

Reply:
```
Traceback (most recent call last):
  File "translate.py", line 6, in <module>
    main()
  File "/OpenNMT-py/onmt/bin/translate.py", line 54, in main
    translate(opt)
  File "/OpenNMT-py/onmt/bin/translate.py", line 16, in translate
    translator = build_translator(opt, logger=logger, report_score=True)
  File "/OpenNMT-py/onmt/translate/translator.py", line 32, in build_translator
    fields, model, model_opt = load_test_model(opt)
  File "/OpenNMT-py/onmt/model_builder.py", line 85, in load_test_model
    map_location=lambda storage, loc: storage)
  File "/venv/lib/python3.6/site-packages/torch/serialization.py", line 584, in load
    return _load(opened_zipfile, map_location, pickle_module, **pickle_load_args)
  File "/venv/lib/python3.6/site-packages/torch/serialization.py", line 842, in _load
    result = unpickler.load()
AttributeError: Can't get attribute '_feature_tokenize' on <module 'onmt.inputters.text_dataset' from '/OpenNMT-py/onmt/inputters/text_dataset.py'>
```

Reply:
That'w what I thought. I'll keep it as it is then